src/Metadata/ClassMetadata.php

```diff
@@ -84,6 +84,13 @@ class ClassMetadata extends MergeableClassMetadata
      */
     public $isMap = false;
 
+    /**
+     * @internal
+     *
+     * @var bool
+     */
+    public $isEnum = false;
```

Comment:
see e.g. `isMap` in this class. It needs to implement the same behaviour for isEnum. Now the caching does not work